tfjs-backend-webgpu/src/backend_webgpu.ts

```diff
@@ -78,6 +78,10 @@ export interface WebGPUTimingInfo extends TimingInfo {
   downloadWaitMs: number;
 }
 
+// Empirically determined constant used to determine size threshold for handing
+// off execution to the CPU.
+const CPU_HANDOFF_SIZE_THRESHOLD = 128;
```

Comment:
Just curious - did you test this constant? Is 128 optimal? Is it approximately the same across different ops?

Reply:
No, I didn't have any data about if 128 is good enough for webgpu. 128 is from WebGL backend.

And, I think this fallback may help unittest or benchmark test. But not real model. Below is my explanation for this. 
If run by yarn test,  "this.tensorMap.get(input.dataId).bufferInfo.buffer == null" will  always be true. This means when we run single unit test, we have a chance to fallback to CPU and get a better performance (So does benchmark test).
When run posnet(though it failed after a short time due to clip not implemented), "this.tensorMap.get(input.dataId).bufferInfo.buffer == null" always return false.  This means, for posnet model, it has no chance to fallback to CPU.

I am not sure if this is the expected behavioural. And did you notice this on WebGL (this.texData.get(input.dataId).texture == null)?

Reply:
@axinging Hmm, it is not expected for the buffer to always be null in the context of a model. Let me investigate this. You're right that if that's the case, the model does not have a chance to benefit from cpu forwarding.

Reply:
Thanks @annxingyuan. And I have another thought about this issue.
For a model with three ops: op1 => op2 => op3. op1's output is op2's input. So does op2 and op3. op1 and op3 run in GPU mode, op2 fallback to CPU mode.

In order to fallback to CPU,  op1's GPU data should be copied into CPU for op2. When op2 is done, op2's CPU data should be copied into GPU for op3. 

For this simple scenario,  there are two copies: one from CPU to GPU, another is from GPU to CPU (And in TFJS's code, I didn't find the copies when fallback. Am I missing something?). We know that copies between CPU and GPU are really time consuming. We should evaluate the fallback mechanism with all these copies.